lib/bundler/dsl.rb

```diff
@@ -332,7 +332,20 @@ def normalize_options(name, version, opts)
       install_if = @install_conditionals.dup
       install_if.concat Array(opts.delete("install_if"))
       install_if = install_if.reduce(true) do |memo, val|
-        memo && (val.respond_to?(:call) ? val.call : val)
+        memo && case val
+                when TrueClass, FalseClass, NilClass
+                  val
+                when Proc
+                  val.call
+                when String
+                  begin
+                    eval(val, TOPLEVEL_BINDING) # rubocop:disable Lint/Eval
+                  rescue
+                    true
+                  end
+                else
+                  true
+                end
```

Comment:
we didn't support evaling strings before, so I don't think this shouldn't be in a random cleanup—it should be in a PR, with a changelog entry and an addition to the docs for `install_if`. also I feel like an indented case statement like this is gross, can you extract the evaluation into a method?